src/realm_dart_sync.cpp

```diff
@@ -30,22 +30,25 @@ RLM_API void realm_dart_http_request_callback(realm_userdata_t userdata, const r
         std::vector<realm_http_header_t> headers_vector;
     } buf;
 
-    realm_http_request_t request_copy = request; // copy struct
-
     buf.url = request.url;
-    request_copy.url = buf.url.c_str();
     buf.body = std::string(request.body, request.body_size);
-    request_copy.body = buf.body.data();
 
     buf.headers_vector.reserve(request.num_headers);
     for (size_t i = 0; i < request.num_headers; i++) {
         auto [it, _] = buf.headers.emplace(request.headers[i].name, request.headers[i].value);
         buf.headers_vector.push_back({ it->first.c_str(), it->second.c_str() });
     }
-    request_copy.headers = buf.headers_vector.data();
 
     auto ud = reinterpret_cast<realm_dart_userdata_async_t>(userdata);
-    ud->scheduler->invoke([ud, request_copy = std::move(request_copy), buf = std::move(buf), request_context]() {
+    ud->scheduler->invoke([ud, request = std::move(request), buf = std::move(buf), request_context]() {
```

Comment:
I don't like we are invalidating a `const` argument here. I suggest we go with #924 where we copy the argument and then we move the local copy

Reply:
Resolving this in favor of #924